### ISSUE TITLE
Incorporate new AWS FSx persistent features into aws_fsx_lustre_file_system resource

### ISSUE THREAD

**Issue by aclaypool** [terraform-providers/terraform-provider-aws#12181]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

AWS has released persistent file systems for AWS FSx Lustre.  Current terraform provider support is only for the "Scratch" file systems.  If aws_fsx_lustre_file_system could be updated to support specifying a persistent file system and any additional required resources.

### New or Affected Resource(s)

aws_fsx_lustre_file_system

### Potential Terraform Configuration

```hcl
resource "aws_fsx_lustre_file_system" "fsx" {
   import_path                   = "s3://path"
   storage_capacity           = 3600
   subnet_ids                     = ["..."] . #now supports more than one
   security_group_ids        = ["..."]
   weekly_maintenance_start_time = "5:04:00"
   deployment_type           "persistent_1|scratch_1|scratch_2"
   encryption_key             = "alias of encryption key"
}
```

### References

https://aws.amazon.com/about-aws/whats-new/2020/02/amazon-fsx-lustre-launches-persistent-file-systems-scratch-file-systems-supports-kms-encryption-in-transit/

https://docs.aws.amazon.com/cli/latest/reference/fsx/create-file-system.html

https://docs.aws.amazon.com/fsx/latest/APIReference/API_CreateFileSystem.html

**Comment by mtpdt** [comment]
I've opened a PR that adds this feature. It's missing the encryption_key field that you requested, but that can be easily added in a subsequent PR.

**Comment by bflad** [comment]
The `deployment_type` handled has been released in [version 3.0.0 of the Terraform AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/v3.0.0/CHANGELOG.md). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.

For further feature requests or bug reports with this functionality, please create a [new GitHub issue](https://github.com/terraform-providers/terraform-provider-aws/issues/new/choose) following the template for triage. Thanks!
